cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -264,6 +264,13 @@ macro(swift_common_unified_build_config product)
   endif()
 endmacro()
 
+# Common cmake project config for compilers.
+macro(swift_common_cxx_compiler_setup)
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
+    set(SWIFT_COMPILER_IS_MSVC_LIKE TRUE)
+  endif()
+endmacro()
```

Comment:
Not sure I see the point in the macro here.  Its a pretty self-contained condition, and I dont think that there are many other instances of aliasing like this that we would need.  Why not inline the single site?